Support Native Kafka Libraries In Docker Image

We are unable to build our docker image due to an issue with the native Kafka library
```shell
#13 87.94     Running setup.py install for confluent-kafka: started
#13 88.78     Running setup.py install for confluent-kafka: finished with status 'error'
#13 88.78     ERROR: Command errored out with exit status 1:
#13 88.78      command: /usr/local/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-xfh5o3vb/confluent-kafka_5197dce6d9024129bbc77ece41a433b4/setup.py'"'"'; __file__='"'"'/tmp/pip-install-xfh5o3vb/confluent-kafka_5197dce6d9024129bbc77ece41a433b4/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-r8_slxkg/install-record.txt --single-version-externally-managed --user --prefix= --compile --install-headers /home/lfh/.local/include/python3.9/confluent-kafka
#13 88.78          cwd: /tmp/pip-install-xfh5o3vb/confluent-kafka_5197dce6d9024129bbc77ece41a433b4/
#13 88.78     Complete output (49 lines):
#13 88.78     running install
#13 88.78     running build
#13 88.78     running build_py
#13 88.78     creating build
#13 88.78     creating build/lib.linux-x86_64-3.9
#13 88.78     creating build/lib.linux-x86_64-3.9/confluent_kafka
#13 88.78     copying src/confluent_kafka/deserializing_consumer.py -> build/lib.linux-x86_64-3.9/confluent_kafka
#13 88.78     copying src/confluent_kafka/error.py -> build/lib.linux-x86_64-3.9/confluent_kafka
#13 88.78     copying src/confluent_kafka/serializing_producer.py -> build/lib.linux-x86_64-3.9/confluent_kafka
#13 88.78     copying src/confluent_kafka/__init__.py -> build/lib.linux-x86_64-3.9/confluent_kafka
#13 88.78     creating build/lib.linux-x86_64-3.9/confluent_kafka/serialization
#13 88.78     copying src/confluent_kafka/serialization/__init__.py -> build/lib.linux-x86_64-3.9/confluent_kafka/serialization
#13 88.78     creating build/lib.linux-x86_64-3.9/confluent_kafka/admin
#13 88.78     copying src/confluent_kafka/admin/__init__.py -> build/lib.linux-x86_64-3.9/confluent_kafka/admin
#13 88.78     creating build/lib.linux-x86_64-3.9/confluent_kafka/kafkatest
#13 88.78     copying src/confluent_kafka/kafkatest/verifiable_client.py -> build/lib.linux-x86_64-3.9/confluent_kafka/kafkatest
#13 88.78     copying src/confluent_kafka/kafkatest/verifiable_producer.py -> build/lib.linux-x86_64-3.9/confluent_kafka/kafkatest
#13 88.78     copying src/confluent_kafka/kafkatest/verifiable_consumer.py -> build/lib.linux-x86_64-3.9/confluent_kafka/kafkatest
#13 88.78     copying src/confluent_kafka/kafkatest/__init__.py -> build/lib.linux-x86_64-3.9/confluent_kafka/kafkatest
#13 88.78     creating build/lib.linux-x86_64-3.9/confluent_kafka/schema_registry
#13 88.78     copying src/confluent_kafka/schema_registry/schema_registry_client.py -> build/lib.linux-x86_64-3.9/confluent_kafka/schema_registry
#13 88.78     copying src/confluent_kafka/schema_registry/error.py -> build/lib.linux-x86_64-3.9/confluent_kafka/schema_registry
#13 88.78     copying src/confluent_kafka/schema_registry/json_schema.py -> build/lib.linux-x86_64-3.9/confluent_kafka/schema_registry
#13 88.78     copying src/confluent_kafka/schema_registry/protobuf.py -> build/lib.linux-x86_64-3.9/confluent_kafka/schema_registry
#13 88.78     copying src/confluent_kafka/schema_registry/__init__.py -> build/lib.linux-x86_64-3.9/confluent_kafka/schema_registry
#13 88.78     copying src/confluent_kafka/schema_registry/avro.py -> build/lib.linux-x86_64-3.9/confluent_kafka/schema_registry
#13 88.78     creating build/lib.linux-x86_64-3.9/confluent_kafka/avro
#13 88.78     copying src/confluent_kafka/avro/cached_schema_registry_client.py -> build/lib.linux-x86_64-3.9/confluent_kafka/avro
#13 88.78     copying src/confluent_kafka/avro/error.py -> build/lib.linux-x86_64-3.9/confluent_kafka/avro
#13 88.78     copying src/confluent_kafka/avro/__init__.py -> build/lib.linux-x86_64-3.9/confluent_kafka/avro
#13 88.78     copying src/confluent_kafka/avro/load.py -> build/lib.linux-x86_64-3.9/confluent_kafka/avro
#13 88.78     creating build/lib.linux-x86_64-3.9/confluent_kafka/avro/serializer
#13 88.78     copying src/confluent_kafka/avro/serializer/message_serializer.py -> build/lib.linux-x86_64-3.9/confluent_kafka/avro/serializer
#13 88.78     copying src/confluent_kafka/avro/serializer/__init__.py -> build/lib.linux-x86_64-3.9/confluent_kafka/avro/serializer
#13 88.78     running build_ext
#13 88.78     building 'confluent_kafka.cimpl' extension
#13 88.78     creating build/temp.linux-x86_64-3.9
#13 88.78     creating build/temp.linux-x86_64-3.9/tmp
#13 88.78     creating build/temp.linux-x86_64-3.9/tmp/pip-install-xfh5o3vb
#13 88.78     creating build/temp.linux-x86_64-3.9/tmp/pip-install-xfh5o3vb/confluent-kafka_5197dce6d9024129bbc77ece41a433b4
#13 88.78     creating build/temp.linux-x86_64-3.9/tmp/pip-install-xfh5o3vb/confluent-kafka_5197dce6d9024129bbc77ece41a433b4/src
#13 88.78     creating build/temp.linux-x86_64-3.9/tmp/pip-install-xfh5o3vb/confluent-kafka_5197dce6d9024129bbc77ece41a433b4/src/confluent_kafka
#13 88.78     creating build/temp.linux-x86_64-3.9/tmp/pip-install-xfh5o3vb/confluent-kafka_5197dce6d9024129bbc77ece41a433b4/src/confluent_kafka/src
#13 88.78     gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -DTHREAD_STACK_SIZE=0x100000 -fPIC -I/usr/local/include/python3.9 -c /tmp/pip-install-xfh5o3vb/confluent-kafka_5197dce6d9024129bbc77ece41a433b4/src/confluent_kafka/src/Admin.c -o build/temp.linux-x86_64-3.9/tmp/pip-install-xfh5o3vb/confluent-kafka_5197dce6d9024129bbc77ece41a433b4/src/confluent_kafka/src/Admin.o
#13 88.78     In file included from /tmp/pip-install-xfh5o3vb/confluent-kafka_5197dce6d9024129bbc77ece41a433b4/src/confluent_kafka/src/Admin.c:17:
#13 88.78     /tmp/pip-install-xfh5o3vb/confluent-kafka_5197dce6d9024129bbc77ece41a433b4/src/confluent_kafka/src/confluent_kafka.h:66:2: error: #error "confluent-kafka-python requires librdkafka v1.6.0 or later. Install the latest version of librdkafka from the Confluent repositories, see http://docs.confluent.io/current/installation.html"
#13 88.78        66 | #error "confluent-kafka-python requires librdkafka v1.6.0 or later. Install the latest version of librdkafka from the Confluent repositories, see http://docs.confluent.io/current/installation.html"
#13 88.78           |  ^~~~~
#13 88.78     error: command '/usr/bin/gcc' failed with exit code 1
#13 88.78     ----------------------------------------
#13 88.78 ERROR: Command errored out with exit status 1: /usr/local/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-xfh5o3vb/confluent-kafka_5197dce6d9024129bbc77ece41a433b4/setup.py'"'"'; __file__='"'"'/tmp/pip-install-xfh5o3vb/confluent-kafka_5197dce6d9024129bbc77ece41a433b4/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-r8_slxkg/install-record.txt --single-version-externally-managed --user --prefix= --compile --install-headers /home/lfh/.local/include/python3.9/confluent-kafka Check the logs for full command output.
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,4 +1,23 @@
-FROM python:3.8-alpine
+FROM python:3.9-alpine
+
+ENV LIBRDKAFKA_VERSION v1.6.1
+
+RUN apk update && \
+    apk add --no-cache --virtual .dev-packages build-base curl bash
+
+# installing librdkafka from source as alpine package repositories may lag behind python confluent kafka requirements
+# librdkafka installation procedure is attributed to https://github.com/confluentinc/confluent-kafka-python
+RUN \
+     echo Installing librdkafka && \
+     mkdir -p /usr/src/librdkafka && \
+     cd /usr/src/librdkafka && \
+     curl -LfsS https://github.com/edenhill/librdkafka/archive/${LIBRDKAFKA_VERSION}.tar.gz | \
+         tar xvzf - --strip-components=1 && \
+     ./configure --prefix=/usr --disable-lz4-ext && \
+     make -j && \
+     make install && \
+     cd / && \
+     rm -rf /usr/src/librdkafka
 
 RUN addgroup -S lfh && adduser -S lfh -G lfh -h /home/lfh
 
@@ -11,5 +30,10 @@ COPY --chown=lfh:lfh ./setup.py setup.py
 COPY --chown=lfh:lfh ./README.md README.md
 COPY --chown=lfh:lfh ./logging.yaml logging.yaml
 RUN pip install --user -e .
+
+USER root
+RUN apk del .dev-packages
+
+USER lfh
 
 CMD ["python", "/home/lfh/pyconnect/main.py"]

